### ISSUE TITLE
Roles and privileges

### ISSUE THREAD

**Issue by wion** [textpattern/textpattern.github.io#48]
I'm mostly done with the new [Roles and privileges](http://docs.textpattern.io/administration/user-roles-and-privileges) doc. I've made some observations for possible consideration against core and Textpacks. [See the four items listed here](http://forum.textpattern.com/viewtopic.php?pid=295136#p295136).

**Comment by rwetzlmayr** [comment]
There is no **Administrator** privilege level in Textpattern AFAIK. What's the purpose of that column in the privilege overview table?

**Comment by wion** [comment]
@rwetzlmayr, I added it originally because there was no explanation that distinguished an administrator from a publisher, and it's a question I've had from clients, actually, when I handed over the website, so there's evidently a gap in the recorded docs that could be filled. 

Though an administrator is not a publishing role, as the doc makes clear, it is an account type, as the doc also makes clear (along with the None/"frozen" account type, having no admin-side privileges/access at all), and it's perhaps useful to distinguish that type from the suite of publishing/collaborator role types. 

But the doc has advanced since first started, and it's probably clear enough at this point that removing the column wouldn't hurt anything. I'd vote leaving it in though since the "None" account is also not a publishing role, per se, but still an option. Might as well cover the board.

Any second motions either way, @Bloke, @philwareham ?

**Comment by Bloke** [comment]
`Administrator` isn't an option from the admin side. It seems you are referring to the hosting platform's account, yes? The one you FTP with. If so, yes it's sort of related insofar as you need such an account to be able to install and perform maintenance actions, so it deserves a mention somewhere, as long as it's clear that it's not configurable from inside Textpttern.

But I agree with Robert that since it's not a core privilege level (unlike `None`, which is), it shouldn't feature in the table. The `Publisher` role has Full access to everything on the admin side, including User management, so that column needs correcting.

**Comment by Bloke** [comment]
Oh, incidentally, the Preferences role system is changing in 4.6 to make it easier for plugins to use it:

In 4.5 there's one privilege level `prefs` that is accessible to only Publishers and Managing Editors.

In 4.6 it's much finer-grained:

    'prefs'                       => '1,2,3,4,5,6',
    'prefs.edit'                  => '1,2',
    'prefs.site'                  => '1,2',
    'prefs.admin'                 => '1,2',
    'prefs.publish'               => '1,2',
    'prefs.feeds'                 => '1,2',
    'prefs.custom'                => '1,2',
    'prefs.comments'              => '1,2',

That means only Publishers and MEs can see the core prefs areas, but the entire prefs panel is open to anyone so that plugins can add their own twisty pref groups there instead of requiring to write a dedicated panel under the Extensions area.

I've done this with smd_article_stats, adding a new priv level `prefs.smd_article_stats` and assigning that to (I think, from memory) Publisher, ME and CE roles. Saves a tonne of code as I can just set up an array of prefs and widgets and let the core draw them all and manage access to that panel, instead of having to do it by hand.

**Comment by wion** [comment]
The Publisher role's limitation reflected in the table was only in relation to the administrator -- can't delete the admin's account. This is explained in the Publisher's section under the table. But yeah, if we remove the Administrator column, then that would be changed since the administrator context would be removed. Certainly.

Okay, then, removing column and adjusting context. ;)

@Bloke: Thanks for the heads up about 4.6. I'll have to think about that before rolling into docs just yet.

**Comment by wion** [comment]
Btw, a question about 4.6 and docs, which I've brought up before but didn't get a response too... We can't get too ahead of ourselves in docs about 4.6 if 4.6 is still many months away. So, our we talking about 3 months? Six? A year? More? If 4.6 is six months or more away, then docs should lean to reflecting 4.5.7 for while longer.

**Comment by rwetzlmayr** [comment]
I think docs should always reflect the **current** stable version. In the wiki, some pages have a "Genealogy" section to document version-specific differences.

**Comment by wion** [comment]
I tend to agree about docs reflecting what release is available, though a few reasonable footnotes are okay, such as how a preference might be relabeled or whatever. But descriptions of functional differences should not be made, no.

I won't be maintaining any genealogy stuff. It's just needless doc bloat in my opinion. Somebody else can do that if they want in tag pages, or whatever, but not me.

**Comment by Bloke** [comment]
Some projects keep docs for different versions online so that users who have not upgraded still have the docs available until they do upgrade. Is that a possibility in Jekyll/Git pages? Like tagging a repo with a 'release' version string or whatever to show you the state of affairs at that point in time?

**Comment by wion** [comment]
> Some projects keep docs for different versions online so that users who have not upgraded still have the docs available until they do upgrade.

Isn't that really when the platform is undergoing radical changes? A new version is spinning off the old and the two are run in parallel until the old can be dropped like a spent rocket stage? That's what I recall with Drupal, for example. In that case you do need two versions of docs.

Otherwise branching is probably a lot of redundancy for little value. And what about the standpoint -- communicated from various Txp properties (and what I've been doing in docs so far too) -- that **users should  update their installations for security reasons**? Keeping a bunch of doc branches available for laggards gives a different message.

IMO, if we keep docs focused on the current context (as they should be), with a few footnote insights about the future, then we are doing the most justice for the end-users who bother to make an effort. We're not only providing the help they need with the latest stable tool, but giving them tasty glimpses of what's coming to (an 'audience retention' technique).

Outside of the tag reference, I'm not sure docs have ever been used much anyway, and that's good, it means the system is relatively intuitive. (Or it means docs haven't been any good, but that's what this new effort is about, hopefully, making them better.) Unless there are going to be radical changes to core, the current docs will only need a few page adjustments over time. Duplicating branches seems like overkill, and even detrimental because people will start linking to old unsupported pages and that ruins all the good linking were trying to build, not just or SEO, but UX across platforms.

2 cents

**Comment by wion** [comment]
In any case, I've update the page. If all is satisfactory, we may close this issue?

**Comment by wion** [comment]
Thinking about this discussion more... It occurs to me that WikiMedia did offer a feature that seems obviously useful now but that we never really used (well, I tried) -- Talk pages. Essentially the backside of the doc, or bar napkin.  For writing down any kind of notes needed, past or future. A good example of it's use, is putting the kind of thing @Bloke did here: https://github.com/textpattern/textpattern.github.io/issues/48#issuecomment-146476397. Then it's right in context of the doc, but not immediately visible to the user looking at current build documentation. When new version is released, it's easy for editor to flip the page and find the relevant notes to guide doc editing.

These issues are kind of like that, but unfortunately not tied to a given doc very well. :{
Sounds like an opportunity for a Jekyll extension. :}

**Comment by philwareham** [comment]
Unfortunately you can't run Jekyll extensions on the GitHub hosted platform (only if you self-host), so that's not an option. It's the same reason I can't get a satisfactory breadcrumb widget working. :(

**Comment by wion** [comment]
In light of the newly revealed 4.6 roadmap, the tentative release timeframe (early 2016), and my _gradual_ progress around other variables of living, it's fair to say I'll be editing docs in more of a 4.6 direction than current 4.5.7, because by the time I'm done, it will be about 4.6 release time.

**Comment by philwareham** [comment]
Agreed. The plan is to get Textpattern 4.6 into beta at the start of January, then use the time between then and release to work on the docs site styling so it launches around the same time as the release. Then work on .com site between 4.6 and 4.7 a the earlier the better.i also want to release 4.7 during 2016 if possible (it's not such an ambitious scope as 4.6 was).
